### PR TITLE
Add linear unlocking functionality for funds held in multisig actors

### PR DESCRIPTION
**Pull request by jesseclay**
This PR adds the ability to put a linear unlock on funds held in multisig actors by passing in a particular unlock period (in blocks from time of account creation -- each block estimated to be 30 seconds). This will be used for Filecoin investor account unlocking, however this could also be used for other use-cases wishing to keep funds locked until a certain block height. With this function the account holder(s) are able to transfer up to and including all funds that have unlocked at that point. Any attempt to transfer more than that will fail.

If the user wishes to not use this functionality at all they can just pass `0` to the initialization function for `unlockDuration`, then no unlock will be used and all funds will be spendable immediately.

### CODE REVIEW

**Comment by jesseclay** [comment]
and a big thanks to @whyrusleeping for the help on this PR.

**Comment by anorth** [comment]
> This will be used for Filecoin investor account unlocking

How does this relate to the ideas expressed in https://github.com/filecoin-project/specs/issues/391? cc @decentralion

**Comment by decentralion** [comment]
As an implementation of unlocking vested funds, this sounds good to me -- I like that the funds progressively unlock but it still requires a user action to transfer them out.

For the calculation of available FIL in #391 - it looks like we would have a separate multisig for every vestor's vest (one investor could have multiple multisigs if they bought with multiple vesting periods). So it wouldn't make sense to sum over all these contracts every time we are caulculating available FIL. However, the whole vesting schedule is known in advance, so we could just hardcode the vesting schedule for calculation purposes independently from the mechanism that actually vests the FIL. It would admit the possibility that these two desynchronize, but if we are careful it won't happen (and the complexity is not that high).

File: actors.md [hunk 1]
```diff
@@ -1440,6 +1448,9 @@ func Propose(to Address, value TokenAmount, method String, params Bytes) UInt {
 	self.Transactions.Append(tx)
 
 	if self.Required == 1 {
+		if !self.canSpend(value) {
```
Comment (whyrusleeping) [inline actors.md]: `tx.Value` here

**Comment by dignifiedquire** [comment]
needs rebasing on master

File: actors.md (hunk 1 above)
Comment (jesseclay) [reply actors.md]: https://github.com/filecoin-project/specs/pull/416/commits/36a09c037300206aeef0cb9df415aa7489613ae8

File: actors.md [hunk 2]
```diff
@@ -1560,6 +1571,9 @@ func Approve(txid UInt) {
 	tx.Approved.Append(msg.From)
 
 	if len(tx.Approved) >= self.Required {
+		if !self.canSpend(tx.Value) {
+			Fatal("transaction amount exceeds available")
+		}
 		tx.RetCode = Send(tx.To, tx.Value, tx.Method, tx.Params)
```
Comment (jesseclay) [inline actors.md]: @whyrusleeping shall we change this to vm.Send to remain consistent with 1528?

**Comment by jesseclay** [comment]
@dignifiedquire should be up to date now: https://github.com/filecoin-project/specs/pull/416/commits/ff27875c57d094d92c7862fd607c2e9f0e1ecc43

File: actors.md (hunk 2 above)
Comment (whyrusleeping) [reply actors.md]: Yeah, thats probably the right thing to do.

Comment (jesseclay) [reply actors.md]: https://github.com/filecoin-project/specs/pull/416/commits/1428d0f1e46a0f78376ff30d78adec279465ceef

**Comment by jesseclay** [comment]
@dignifiedquire @whyrusleeping all comments addressed now. LMK if you'd like me to merge or just feel free.

**Pull request merged**